proposed/event-dispatcher.md

```diff
@@ -124,145 +86,94 @@ A Listener Provider is a service object responsible for determining what Listene
 * Extracting some information from an object referenced by the Event, such as an Entity, and calling pre-defined lifecycle methods on that object.
+* Delegating its responsibility to one or more other Listener Providers using some arbitrary logic.
 
 Or some combination of those, or some other mechanism as desired.
 
 All Listeners returned by a Listener Provider MUST be type-compatible with the Event.  That is, calling `$listener($event)` MUST NOT produce a `TypeError`.
 
-## Object composition
-
-A Notifier or Processor SHOULD compose a Listener Provider to determine relevant listeners.  It is RECOMMENDED that a Listener Provider be implemented as a distinct object from the Notifier or Processor but that is NOT REQUIRED.
-
-It is RECOMMENDED that a Notifier and Processor be implemented as distinct objects.  However, an object MAY implement both interfaces if deemed appropriate.
-
-## Interfaces
-
-### General event interfaces
+Listener Providers MUST treat parent types identically to the Event's own type when determining listener applicability.  That is, in the following case:
 
 ```php
-namespace Psr\EventDispatcher;
+class A {}
 
-/**
- * Marker interface indicating an event instance.
- *
- * Event instances may contain zero methods, or as many methods as they
- * want. The interface MUST be implemented, however, to provide type-safety
- * to both listeners as well as listener providers.
- */
-interface EventInterface
-{
-}
-```
+class B extends A {}
 
-```php
-namespace Psr\EventDispatcher;
+$b = new B();
 
-/**
- * Mapper from an event to the listeners that are applicable to that event.
- */
-interface ListenerProviderInterface
-{
-    /**
-     * @param EventInterface $event
-     *   An event for which to return the relevant listeners.
-     * @return iterable[callable]
-     *   An iterable (array, iterator, or generator) of callables.  Each
-     *   callable MUST be type-compatible with $event.
-     */
-    public function getListenersForEvent(EventInterface $event) : iterable;
-}
+function listener(A $event) {}
 ```
 
-### Message Interfaces
+A Listener Provider MUST treat `listener()` as an applicable listener for `$b`, as it is type compatible, unless some other criteria prevents it from doing so.
 
-```php
-namespace Psr\EventDispatcher;
+## Object composition
 
+A Dispatcher SHOULD compose a Listener Provider to determine relevant listeners.  It is RECOMMENDED that a Listener Provider be implemented as a distinct object from the Dispatcher but that is NOT REQUIRED.
 
-/**
- * This is a marker interface used to identify message events.
- */
-interface MessageInterface extends EventInterface
-{
-}
-```
+## Interfaces
 
 ```php
 namespace Psr\EventDispatcher;
 
 /**
- * Defines a notifier for message events.
+ * Defines a dispatcher for events.
  */
-interface MessageNotifierInterface
+interface EventDispatcherInterface
 {
     /**
-     * Notify listeners of a message event.
+     * Provide all listeners with an event to process.
```

Comment:
`s/Provide all/Provide all relevant/`, as not _every_ listener will be notified, only those capable of processing the event.

Reply:
Dur, I'll have to change that in the package, too.  Need to leave a note to myself somewhere.